src/main/java/bio/terra/app/configuration/OpenIDConnectConfiguration.java

```diff
@@ -0,0 +1,144 @@
+package bio.terra.app.configuration;
+
+import bio.terra.common.exception.ServiceInitializationException;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+@EnableConfigurationProperties
+@ConfigurationProperties(prefix = "oidc")
+public class OpenIDConnectConfiguration {
+  private static final String OIDC_METADATA_URL_SUFFIX = ".well-known/openid-configuration";
+  private final RestTemplate oidcRestTemplate;
+
+  private String schemeName;
+  private String authorityEndpoint;
+  private String clientId;
+  private String clientSecret;
+  private String extraAuthParams;
+  private boolean addClientIdToScope;
+
+  private String authorizationEndpoint;
+  private String tokenEndpoint;
+
+  @Autowired
+  public OpenIDConnectConfiguration() {
+    this.oidcRestTemplate = new RestTemplate(new HttpComponentsClientHttpRequestFactory());
+    MappingJackson2HttpMessageConverter converter = new MappingJackson2HttpMessageConverter();
+    converter.setSupportedMediaTypes(Collections.singletonList(MediaType.ALL));
+    oidcRestTemplate.setMessageConverters(List.of(converter));
+  }
+
+  @PostConstruct
+  @VisibleForTesting
+  void init() {
+    if (!StringUtils.isEmpty(getAuthorityEndpoint())) {
+      HttpHeaders headers = new HttpHeaders();
+      headers.setContentType(MediaType.APPLICATION_JSON);
+
+      ResponseEntity<OpenIDProviderMetadata> metadataConfig =
```

Comment:
I see - so you're actually calling `GET https://<authority>/.well-known/openid-configuration` at initialization time to get the authorize and token endpoints. That's probably a good idea as it validates the authority endpoint too. I may do the same on the Scala side..

Reply:
Yeah, that ended up simplifying things quite a bit